packages/rich-text/src/create.js

```diff
@@ -306,15 +305,14 @@ function collapseWhiteSpace( string ) {
 	return string.replace( /[\n\r\t]+/g, ' ' );
 }
 
-const ZWNBSPRegExp = new RegExp( ZWNBSP, 'g' );
-
 /**
- * Removes padding (zero width non breaking spaces) added by `toTree`.
+ * Removes reserved characters used by rich-text (zero width non breaking spaces added by `toTree` and object replacement characters).
  *
  * @param {string} string
  */
-function removePadding( string ) {
-	return string.replace( ZWNBSPRegExp, '' );
+export function removeReservedCharacters( string ) {
+	//with the global flag, note that we should create a new regex each time OR reset lastIndex state.
+	return string.replace( /[\uFEFF\uFFFC]/gu, '' );
```

Comment:
It would be good to use `OBJECT_REPLACEMENT_CHARACTER` and `ZWNBSP` form './special-characters', both of which are already imported above.